### ISSUE TITLE
NPE when elasticSearch.unmarshallComponents config is missing

### ISSUE THREAD

**Issue by micke-a** [mstein/elasticsearch-grails-plugin#62]
HI

Problem:
Just started using latest 0.20.6.1-SNAPSHOT and got NPEs whenever there was a search hit.
Turned out it was due to new elasticSearch.unmarshallComponents config element.

Fix:
Add as 'true' to default parameters?

Cheers,
 micke

**Comment by mstein** [comment]
Actually the `unmarshallComponents` property is set to `true` in the `DefaultElasticSearch.groovy` file in the `grails-app/conf/` folder of the installed plugin. Could you check if that is the case in your installation of the plugin?

**Comment by micke-a** [comment]
Hi

I've got this in my app

``` groovy
elasticSearch{
    cluster.name = "app-"+grails.util.Environment.current
    client.mode = 'local'
    index.store.type = 'simplefs'
    path.data = "/tmp/es-data"
    defaultExcludedProperties = ['password','email','passwd']
    unmarshallComponents = true
}
```

I've got an admin screen where I display al the configration properties by creating a flatten:ed version of it, and in there I can only see my confg values when searching for elasticSearch....

Not quite sure what is going on, any ideas? Not causing me any issues at the moment though. Recently looked at the code for the async mail plugin which does similar config stuff and that may have a better way of merging different configs.

Cheers
